File: 2740/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(findValueOfPartition([]int{1,3,2,4}))
	fmt.Println(findValueOfPartition([]int{84,11,100,100,75}))

}

func findValueOfPartition(nums []int) int {
	sort.Ints(nums)
	ans := 100000000000
	for i := 0; i <= len(nums)-2; i++ {
		d := abs(nums[i], nums[i+1])
		ans = min(ans, d)
	}
	return ans
}

func abs(a, b int) int {
	t := a - b
	if t < 0 {
		return -t
	}
	return t
}

File: 2740/main.rs
struct Solution{}

impl Solution {
    pub fn find_value_of_partition(mut nums: Vec<i32>) -> i32 {
        let mut ans = 1000000000;
        let n = nums.len();
        nums.sort();
        for i in 0..n - 1 {
            ans = std::cmp::min((nums[i] - nums[i+1]).abs(), ans);
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::find_value_of_partition(vec![1,3,2,4]));
    println!("{}", Solution::find_value_of_partition(vec![84,11,100,100,75]));

}